### ISSUE TITLE
Give clere error message when a method matches the class name that has an explicit __construct method

### ISSUE THREAD

**Issue by bkdotcom** [phpmd/phpmd#231]
I got a false positive on this one.

```php
<?php
class Foo
{

    /**
     * This is the constructor
     */
    public function __construct()
    {
    }

    /**
     * this is not a constructor

     * @return boolean
     */
    public function foo()
    {
        return false;
    }
}
```

The constructor is `__construct()`  `foo()` is just a regular 'ol function and as such should not raise said error

**Comment by ravage84** [comment]
@bkdotcom or @internalsystemerror, if you want, you could create a PR with a new rule that covers this.

I would name it like ``ClassNamedMethod`` and would place it into the [naming folder](https://phpmd.org/rules/index.html#naming-rules).
